client/search.jsx

```diff
@@ -0,0 +1,126 @@
+
+import fetch from 'node-fetch';
+import React from 'react';
+import Loader from './loader';
+import NavBar from './navBar';
+class Search extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userInput: '',
+      hotelList: [],
+      searchButtonClicked: '',
+      hotelThumbnails: [],
+      ratingFilter: '',
+      isLoading: false,
+      userInputError: false
+    };
+    this.handleSearchClick = this.handleSearchClick.bind(this);
+    this.handleUserInput = this.handleUserInput.bind(this);
+    this.handleBackClick = this.handleBackClick.bind(this);
+    this.ratingFilter = this.ratingFilter.bind(this);
+  }
+
+  ratingFilter(event) {
+    this.setState({ ratingFilter: event.target.value });
+  }
+
+  handleUserInput(event) {
+    this.setState({ userInput: event.target.value });
+  }
+
+  handleBackClick() {
+    this.setState({ userInput: '', hotelList: [], searchButtonClicked: '', hotelThumbnails: [], isLoading: false, userInputError: false });
+  }
+
+  handleSearchClick(event) {
+
+    event.preventDefault();
+    if (!this.state.ratingFilter) {
+      return this.setState({ userInputError: true });
+    }
+    const arr = this.state.userInput;
+    const notAllowed = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '`', '!', '#', '$', '%', '^', '&', '-', '_',
+      '*', '(', ')', '+', '=', '<', '>', ',', '.', '?', ':', ';', '@', '{', '}', '[', ']', '~', '/'];
+
+    for (let i = 0; i < notAllowed.length; i++) {
+      if (arr === '' || arr.includes(notAllowed[i])) {
+        return this.setState({ userInputError: true });
+      }
+    }
+    this.setState({ isLoading: true });
+    fetch(`/api/search/${arr}`)
+      .then(response => {
+        return response.json();
+      })
+      .then(data => {
+        const cityId = data.suggestions[0].entities[0].destinationId;
+        fetch(`/api/search/list/${cityId}/${this.state.ratingFilter}`).then(response => response.json()).then(data => {
+          const hotelList = data.data.body.searchResults.results;
+          for (const hotel of hotelList) {
+            this.state.hotelList.push(hotel.name);
+            this.state.hotelThumbnails.push(hotel.thumbnailUrl);
+          }
```

Comment:
Loop and push is what `Array.prototype.map()` is for. In general, we never mutate objects or arrays that are in state.

Also, why are the hotel thumbnails in a separate list from the hotel objects?